src/plugins/setup.ts

```diff
@@ -167,6 +167,33 @@ export default createPlugin(
 								}
 							}
 
+							/////////////////////////////////////////////////////////////////////
+							progress.report({ message: "Checking LocalStack license..." });
+
+							// If an auth token has just been obtained or LocalStack has never been started,
+							// then there will be no license info to be reported by `localstack license info`.
+							// Also, an expired license could be cached.
+							// Activating the license pre-emptively to know its state during the setup process.
+							await activateLicense(outputChannel);
+							const licenseIsValid = await minDelay(
+								checkIsLicenseValid(outputChannel),
+							);
```

Comment:
Nitpick: the way it currently is, the user will wait until `activateLicense` finished, and then the minimum delay between `checkIsLicenseValid` and 0.5s.

We can streamline both `activateLicense` and `checkIsLicenseValid` by creating a chain with `Promise.then`, and awaiting the result of that. This effectively becomes a "wait 0.5 minimum, or whatever it takes to activateLicense and checkIsLicenseValid together".

```suggestion
							const licenseIsValid = await minDelay(
								activateLicense(outputChannel).then(
									() => checkIsLicenseValid(outputChannel),
								),
							);
```